src/controllers/attachments/index.ts

```diff
@@ -1,32 +1,120 @@
-import { Controller, Get, Route } from 'tsoa'
+import { UploadedFile, Request, Controller, Get, Route, Path, Post, Header, Response } from 'tsoa'
 import { Logger } from 'pino'
+import express from 'express'
 
 import { logger } from '../../lib/logger'
-import Database from '../../lib/db'
+import Database, { Models, Query } from '../../lib/db'
 import type { Attachments } from '../../models'
+import { BadRequst, NotFound } from '../../lib/error-handler'
+
+/**
+ * A successfull reponse
+ * @example { "status": 200, "data": [ { "id": "5a3f54bd-a459-45fb-8e50-9ffdd26ff981", "filename": "Screenshot 2023-03-13 at 15.26.41.png", "created_at": "2023-03-14T11:56:19.832Z" } ] }
+ */
+interface Success {
+  status: 200 | 201
+  message?: string
+  data?: Attachments[] | Attachments
+  headers?: { [k: string]: unknown }
+}
+
+type File = {
+  [k: string]: string
+}
 
 @Route('attachments')
```

Comment:
based off our previous conventions, the route should be `attachment` rather than `attachments`?

Reply:
mmm... happy to change, but it has to be inline with database model. Not sure why we have called plural. Since on row = one file? I might be wrong.

Reply:
it doesn't need to be inline with the database model, I'm using `@Route('capacity')` with a `Demands` model

Reply:
for tsoa yes, you can use any other route, but since the entity in database is **attachments** I would like to stick to it for obvious reasons. If database entity and route are totally different, like here

The whole idea is that TSOA model is inline with database model? Why because then we can utilize typescript for types and definitions, so no need to double up on definitions. At the moment is hard to see the benefit since the database is not fully implemented and we jsut create a connection

This will help us to have something called REST factory (get basic CRUD right of the model) e.g. would like to get template to that level so we just need to write code and use a strongly typed approach. To be fair easier would be to talk through

Reply:
the REST factory idea sounds good, but isn't implemented in this PR. And `@Route('attachments')` -> `@Route('attachment')` doesn't affect anything else. Our other APIs use single form rather than plural
https://github.com/digicatapult/morello-api/blob/cbefd92745b7d57288f935f98898f289e564f50f/src/controllers/scenario/index.ts#L10
![image](https://user-images.githubusercontent.com/40722025/225604494-8dbcaca6-eed6-4025-9a3f-1c4abf946a2c.png)